### ISSUE TITLE
ExpressionExperimentBatchCorrectionServiceTest.testComBatOnEE fails on the CI

### ISSUE THREAD

**Issue by arteymix** [PavlidisLab/Gemma#1181]
The test works fine when run on its own.

**Comment by arteymix** [comment]
Nevermind, it's reproducible. I'll look into it before releasing.

**Comment by arteymix** [comment]
```
java.lang.IllegalStateException: None of the samplesUsed have a value for factor:  ExperimentalFactor Id=3 Name=treatment Type=CATEGORICAL Category=Characteristic Id=77 Category=treatment [http://www.ebi.ac.uk/efo/EFO_0000727] Value=treatment [http://www.ebi.ac.uk/efo/EFO_0000727] (2 factor values) - ensure samples are assigned this factor

	at ubic.gemma.core.datastructure.matrix.ExpressionDataMatrixColumnSort.getBaselineLevels(ExpressionDataMatrixColumnSort.java:172)
	at ubic.gemma.model.expression.experiment.ExperimentalDesignUtils.getBaselineConditions(ExperimentalDesignUtils.java:131)
	at ubic.gemma.core.analysis.preprocess.batcheffects.ExpressionExperimentBatchCorrectionServiceImpl.getDesign(ExpressionExperimentBatchCorrectionServiceImpl.java:384)
	at ubic.gemma.core.analysis.preprocess.batcheffects.ExpressionExperimentBatchCorrectionServiceImpl.comBat(ExpressionExperimentBatchCorrectionServiceImpl.java:175)
	at ubic.gemma.core.analysis.preprocess.batcheffects.ExpressionExperimentBatchCorrectionServiceImpl.comBat(ExpressionExperimentBatchCorrectionServiceImpl.java:159)
	at ubic.gemma.core.analysis.preprocess.batcheffects.ExpressionExperimentBatchCorrectionServiceTest.testComBatOnEE(ExpressionExperimentBatchCorrectionServiceTest.java:114)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.springframework.test.context.junit4.statements.RunBeforeTestMethodCallbacks.evaluate(RunBeforeTestMethodCallbacks.java:74)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.springframework.test.context.junit4.statements.RunAfterTestMethodCallbacks.evaluate(RunAfterTestMethodCallbacks.java:83)
	at org.mockito.internal.junit.JUnitSessionStore$1.evaluateSafely(JUnitSessionStore.java:55)
	at org.mockito.internal.junit.JUnitSessionStore$1.evaluate(JUnitSessionStore.java:43)
	at org.springframework.test.context.junit4.statements.SpringRepeat.evaluate(SpringRepeat.java:72)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:231)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:88)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.springframework.test.context.junit4.statements.RunBeforeTestClassCallbacks.evaluate(RunBeforeTestClassCallbacks.java:61)
	at org.springframework.test.context.junit4.statements.RunAfterTestClassCallbacks.evaluate(RunAfterTestClassCallbacks.java:71)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.run(SpringJUnit4ClassRunner.java:174)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater$1.execute(IdeaTestRunner.java:38)
	at com.intellij.rt.execution.junit.TestsRepeater.repeat(TestsRepeater.java:11)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:35)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:232)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:55)
```

Looks like an issue with the imported design.

**Comment by arteymix** [comment]
Fixed in ef85f1d415c21890aaf40cfa8437f5b644c976dd.

It could also be fixed by rethawing the EE, but instead I fixed how the ED importer populates biomaterial FVs to avoid loading different instances if the EE is not in the session.
